custom_components/enphase_envoy/envoy_reader.py

```diff
@@ -251,8 +356,17 @@ async def _fetch_owner_token_json(self):
             return resp.text
 
     async def _getEnphaseToken(self):
-        self._token = await self._fetch_owner_token_json()
-        _LOGGER.debug("Commissioned Token")
+        # When we have a valid commissioned token, but run into 401's,
+        # we should switch to envoy tokens, as those have more permissions for DHZ accounts
+        if self._token and not self._is_enphase_token_expired(self._token):
+            self.use_envoy_tokens = True
```

Comment:
Not really sure if this is the best way to do it, but this works for me, and allows a normal owner token to be used every time..